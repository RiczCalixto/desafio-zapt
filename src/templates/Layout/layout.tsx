import { P } from 'components/Typography/typography.style'
import * as S from './layout.style'

type LayoutProps = { children: React.ReactNode }

export const Layout = ({ children }: LayoutProps) => (
  <S.GridArea>
    <S.Header>
      <S.Logo src="/img/logo.svg" alt="Imagem de logo da Zapt" />
    </S.Header>
    <S.Main>{children}</S.Main>
    <S.Footer>
      <P>Desafio Frontend Zapt 🧑‍💻</P>
    </S.Footer>
  </S.GridArea>
)
