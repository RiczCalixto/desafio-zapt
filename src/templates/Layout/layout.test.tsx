import { render, screen } from 'utils/test-utils'

import { Layout } from './layout'

describe('<Base />', () => {
  it('should render base template with logo and footer', () => {
    const { container } = render(
      <Layout>
        <h1>Content</h1>
      </Layout>
    )
    const img = screen.getByRole('img', { name: /Imagem de logo da Zapt/i })
    const heading = screen.getByRole('heading', { name: /Content/i })
    const paragraph = screen.getByText(/Desafio Frontend Zapt üßë‚Äçüíª/i)

    expect(img).toHaveAttribute('src', '/img/logo.svg')
    expect(heading).toBeInTheDocument()
    expect(paragraph).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
