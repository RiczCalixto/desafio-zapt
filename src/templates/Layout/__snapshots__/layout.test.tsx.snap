// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render base template with logo and footer 1`] = `
.c5 {
  color: #000000;
  font-size: 1.6rem;
  font-family: Inter,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 400;
  line-height: 2.4rem;
  margin: 0;
}

.c0 {
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: auto 1fr auto;
  grid-template-areas: 'header' 'main' 'footer';
  height: 100%;
}

.c1 {
  grid-area: header;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 1px solid rgba(0,0,0,0.12);
  padding: 5px 0;
}

.c2 {
  width: 10rem;
  height: 3.2rem;
}

.c3 {
  grid-area: main;
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 10px;
  padding-left: 10px;
}

.c4 {
  grid-area: footer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 0;
}

@media (max-width:768px) {
  .c5 {
    font-size: 1.4rem;
  }
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <img
      alt="Imagem de logo da Zapt"
      class="c2"
      src="/img/logo.svg"
    />
  </nav>
  <main
    class="c3"
  >
    <h1>
      Content
    </h1>
  </main>
  <footer
    class="c4"
  >
    <p
      class="c5"
    >
      Desafio Frontend Zapt üßë‚Äçüíª
    </p>
  </footer>
</div>
`
